src-java/kilda-model/src/main/java/org/openkilda/model/PhysicalPort.java

```diff
@@ -0,0 +1,153 @@
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.experimental.Delegate;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.factory.Mappers;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Represents a physical port.
+ */
+@DefaultSerializer(BeanSerializer.class)
+@ToString
+public class PhysicalPort implements CompositeDataEntity<PhysicalPortData> {
+    @Getter
+    @Setter
+    @Delegate
+    @JsonIgnore
+    private PhysicalPortData data;
+
+    /**
+     * No args constructor for deserialization purpose.
+     */
+    private PhysicalPort() {
+        data = new PhysicalPortDataImpl();
+    }
+
+    /**
+     * Cloning constructor which performs deep copy of the entity.
+     *
+     * @param entityToClone the entity to copy entity data from.
+     */
+    public PhysicalPort(@NonNull PhysicalPort entityToClone) {
+        data = PhysicalPortCloner.INSTANCE.deepCopy(entityToClone.getData());
+    }
+
+    @Builder
+    public PhysicalPort(@NonNull SwitchId switchId, int portNumber, @NonNull LagLogicalPort lagLogicalPort) {
+        data = new PhysicalPortDataImpl(switchId, portNumber, lagLogicalPort);
+    }
+
+    public PhysicalPort(@NonNull PhysicalPort.PhysicalPortData data) {
+        this.data = data;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        PhysicalPort that = (PhysicalPort) o;
+        return new EqualsBuilder()
+                .append(getPortNumber(), that.getPortNumber())
+                .append(getSwitchId(), that.getSwitchId())
+                .isEquals();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getPortNumber(), getSwitchId());
+    }
+
+    /**
+     * Defines persistable data of the PhysicalPort.
+     */
+    public interface PhysicalPortData {
+        SwitchId getSwitchId();
+
+        void setSwitchId(SwitchId switchId);
+
+        LagLogicalPort getLagLogicalPort();
+
+        void setLagLogicalPort(LagLogicalPort lagLogicalPort);
+
+        int getPortNumber();
+
+        void setPortNumber(int portNumber);
+    }
+
+    /**
+     * POJO implementation of PhysicalPortData.
+     */
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
+    static final class PhysicalPortDataImpl implements PhysicalPortData, Serializable {
+        private static final long serialVersionUID = 1L;
+        @NonNull SwitchId switchId;
+        int portNumber;
+
+        @ToString.Exclude
+        @EqualsAndHashCode.Exclude
+        @NonNull LagLogicalPort lagLogicalPort;
```

Comment:
But will the setter method update a back reference in LagLogicalPort? I think, no.

Reply:
fixed